From 4faf5244a0fc1166c285a39d56416d1743ae36ea Mon Sep 17 00:00:00 2001
From: David Cumps <david@cumps.be>
Date: Sun, 3 Apr 2016 19:57:51 +0200
Subject: [PATCH] Tweak receiver.cc for Alpine

---
 pdns/receiver.cc | 14 +-------------
 1 file changed, 1 insertion(+), 13 deletions(-)

diff --git a/pdns/receiver.cc b/pdns/receiver.cc
index a2f5591..b25d314 100644
--- a/pdns/receiver.cc
+++ b/pdns/receiver.cc
@@ -374,21 +374,9 @@ static void loadModules()
 }
 
 #ifdef __GLIBC__
-#include <execinfo.h>
 static void tbhandler(int num)
 {
-  L<<Logger::Critical<<"Got a signal "<<num<<", attempting to print trace: "<<endl;
-  void *array[20]; //only care about last 17 functions (3 taken with tracing support)
-  size_t size;
-  char **strings;
-  size_t i;
-  
-  size = backtrace (array, 20);
-  strings = backtrace_symbols (array, size); //Need -rdynamic gcc (linker) flag for this to work
-  
-  for (i = 0; i < size; i++) //skip useless functions
-    L<<Logger::Error<<strings[i]<<endl;
-  
+  L<<Logger::Critical<<"Got a signal "<<num<<" (trace printing hard-disabled)"<<endl;
   
   signal(SIGABRT, SIG_DFL);
   abort();//hopefully will give core
-- 
2.6.3.windows.1

